### PR TITLE
Different small umdl and crawler fixes

### PR DESCRIPTION
**Pull request by adrianreber**
This PR contains three smaller changes:

* Bring some changes from @pypingou's umdl2 branch to the current umdl to correctly handle version and repository creation.
* Handle continents which are not specified in upper case.
* Add python-GeoIP as a RPM dependency to the crawler sub-package.

### CODE REVIEW

File: utility/mm2_crawler [hunk 1]
```diff
@@ -135,7 +135,12 @@ def doit(options, config):
         if any(item.startswith('^') for item in options.continents):
             for item in options.continents:
                 if item.startswith('^'):
-                    continents.remove(item[1:])
+                    try:
+                        continents.remove(item[1:].upper())
+                    except ValueError:
+                        # A continent was specified which is not
```
Comment (ralphbean) [inline utility/mm2_crawler]: Perhaps log a warning here instead of passing silently?

Comment (adrianreber) [inline utility/mm2_crawler]: Yes, you are right. I thought the same when writing that part. I will shorty update this PR and then the output will be like this:

```
WARNING:crawler:Hosts(0/0):Threads(0/0):0:master:Ignoring unknown continent: E
WARNING:crawler:Hosts(0/0):Threads(0/0):0:master:Not in the list of known continents. ['AF', 'AN', 'AS', 'EU', 'NA', 'OC', 'SA', '--']
```

**Comment by adrianreber** [comment]
This fixes:
* https://fedorahosted.org/fedora-infrastructure/ticket/4839
* https://fedorahosted.org/fedora-infrastructure/ticket/4837

**Comment by ralphbean** [comment]
:+1: to merge, here.

**Comment by pypingou** [comment]
:+1: as well, want me to cut a new release with these?

**Comment by adrianreber** [comment]
I have to run some additional tests to make it sure everything works as expected before merging.

**Comment by adrianreber** [comment]
I updated the PR with additional changes from #111 8d499c8af12551697e78816934acbd0ae6ee081e. Now umdl creates the repositories and updates the repomd.xml files correctly.

**Comment by pypingou** [comment]
Still :+1: for me

**Comment by adrianreber** [comment]
@pypingou a new release would be appreciated to get the fixed errors deployed in prod.

**Pull request merged**